Sources/Tracking/CustomerIO.swift

```diff
@@ -78,4 +67,43 @@ public extension CustomerIO {
 
         initializeSharedInstance(with: implementation, diGraph: newDiGraph)
     }
+
+    /**
+     Common initialization method for setting up the shared `CustomerIO` instance.
+     This method is intended to be used by both actual implementations and in tests, ensuring that tests closely mimic the real-world implementation.
+     */
+    private static func initialize(implementation: CustomerIOInstance, diGraph: DIGraph) {
+        initializeSharedInstance(with: implementation, diGraph: diGraph)
+
+        // Handle any unprocessed tasks pending in the background queue.
+        let migrationAssistant = diGraph.dataPipelineMigrationAssistant
+        migrationAssistant.handleQueueBacklog()
+
+        let sdkConfig = diGraph.sdkConfig
+        // automatically add the Logger plugin if logLevel is debug
+        if sdkConfig.logLevel == .debug {
+            CustomerIO.shared.setDebugLogsEnabled(true)
+        }
+        // automatically add the AutoTrackingScreenViews plugin if autoTrackScreenViews is enabled
+        if sdkConfig.autoTrackScreenViews {
+            DataPipeline.shared.analytics.add(plugin: AutoTrackingScreenViews(filterAutoScreenViewEvents: sdkConfig.filterAutoScreenViewEvents, autoScreenViewBody: sdkConfig.autoScreenViewBody))
+        }
+    }
+
+    #if DEBUG
+    // Methods to set up the test environment.
+    // Integration tests use the actual implementation.
+
+    @discardableResult
+    static func setUpSharedInstanceForIntegrationTest(diGraphShared: DIGraphShared, diGraph: DIGraph, autoAddCustomerIODestination: Bool) -> CustomerIOInstance {
```

Comment:
`autoAddCustomerIODestination` should probably have a default value?

Reply:
Updated